File: ruby/helpers/log_in_helper.rb
require 'dotenv'
require_relative '../lib/codec'

module LogInHelper

  def decrypt_credentials
    Dotenv.load
    Codec
        .new(
            key: ENV['credentials_key'],
            file_path: '../encrypted_data/credentials.yml'
        )
        .load_file
  end

  def load_credentials
    credentials = decrypt_credentials
    @user_name = credentials['user_name']
    @password = credentials['password']
  end

end

File: playwrite_js/helpers/log_in_helper.js
import path from 'path';
import dotenv from 'dotenv';

// playwrite_js/.env
dotenv.config({ path: path.resolve(import.meta.dirname, '..', '.env') });

import Codec from '../lib/codec.js';

const codec = new Codec(process.env.CREDENTIAL_KEY, path.resolve(import.meta.dirname,'..','encrypted_data/credentials.json'));
const data = await codec.decryptData();

export function user_name() {
    return data.user_name;
};

export function password() {
    return data.password;
};